### PR TITLE
Remove all unsound tests

### PR DESCRIPTION
**Pull request by elliette**

### CODE REVIEW

**Review by Markzipan — APPROVED** [review]
Thanks for the cleanup!

File: fixtures/_test/example/scopes/main.dart [hunk 1]
```diff
@@ -1,166 +0,0 @@
-// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
-// for details. All rights reserved. Use of this source code is governed by a
-// BSD-style license that can be found in the LICENSE file.
-
```
Comment (nshahan) [inline fixtures/_test/example/scopes/main.dart:4]: This one doesn't have the language version comment like the other examples. Was this actually a test for unsound mode?

File: dwds/test/fixtures/utilities.dart [hunk 2]
```diff
@@ -298,12 +297,11 @@ class TestBuildSettings extends BuildSettings {
 
 class TestCompilerOptions extends CompilerOptions {
   TestCompilerOptions({
-    required NullSafety nullSafety,
     required super.canaryFeatures,
     required List<String> experiments,
     super.moduleFormat = ModuleFormat.amd,
   }) : super(
-          soundNullSafety: nullSafety == NullSafety.sound,
+          soundNullSafety: true,
```
Comment (nshahan) [inline dwds/test/fixtures/utilities.dart:304]: Will there be followup to just remove this flag entirely?

File: webdev/test/e2e_test.dart [hunk 3]
```diff
@@ -279,316 +251,301 @@ void main() {
...
+        configureLogWriter(debug);
+      });
+      test('evaluateInFrame', () async {
+        var openPort = await findUnusedPort();
+        // running daemon command that starts dwds without keyboard input
+        var args = [
+          'daemon',
+          'web:$openPort',
+          '--enable-expression-evaluation',
+          '--null-safety=sound',
```
Comment (nshahan) [inline webdev/test/e2e_test.dart:265]: Is this argument required? Hopefully this will become the default with no flags needed.

**Review by nshahan — APPROVED** [review]

File: fixtures/_test/example/scopes/main.dart (hunk 1 above)
Comment (elliette) [reply fixtures/_test/example/scopes/main.dart:4]: Hmm it looks like this fixture wasn't actually being used in any tests

File: dwds/test/fixtures/utilities.dart (hunk 2 above)
Comment (elliette) [reply dwds/test/fixtures/utilities.dart:304]: Yep, will do a follow up :)

File: webdev/test/e2e_test.dart (hunk 3 above)
Comment (elliette) [reply webdev/test/e2e_test.dart:265]: Hopefully not! Will remove in a follow up

**Pull request merged**